servicetalk-concurrent-api/src/main/java/io/servicetalk/concurrent/api/TimeoutPublisher.java

```diff
@@ -86,7 +87,7 @@ void handleSubscribe(Subscriber<? super T> subscriber, SignalOffloader signalOff
                 signalOffloader, contextMap, contextProvider);
     }
 
-    private static final class TimeoutSubscriber<X> implements Subscriber<X>, Subscription, Runnable {
```

Comment:
A `Runnable` declaration doesn't explain the context under which the `run()` method is to be called. Here `Runnable` is essentially an implementation detail used for an API that the class calls rather than part of the API it offers to callers. That `run()` is a public method also might make it appear that `run()` is part of the API of `TimeoutSubscriber`. Using a direct method reference (and giving the method a useful name) provides better stack traces while providing the same performance and allows removal of an extraneous interface.